packages/validator/src/capability.js

```diff
@@ -635,29 +635,52 @@ class AndMatch {
 }
 
 /**
- * Parses capability `source` using a provided capability `parser`. By default
- * invocation parsing occurs, which respects a capability schema, failing if
- * any non-optional field is missing. If `optional` argument is `true` it will
- * parse capability as delegation, in this case all `nb` fields are considered
- * optional.
+ * @template {string} T
+ * @param {T} match
+ * @param {string} pattern
+ * @returns {T|null}
+ */
+const matchAbility = (match, pattern) =>
+  match === pattern
+    ? match
+    : pattern === '*'
+    ? match
+    : pattern.endsWith('/*') && match.startsWith(pattern.slice(0, -1))
+    ? match
+    : null
+
+/**
+ * @template {string} T
```

Comment:
```suggestion
 * Function is used by the delegated ability parser to resolve `with` resource of the delegated
 * capability. It takes `match` argument corresponding to the `with` of the claimed capability
 * and a `pattern` corresponding to the `with` of the delegated capability. If pattern matches
 * returns `match` back, otherwise returns `null`.
 * @template {string} T
```